internal/otelcollector/config/metric/gateway/config_builder.go

```diff
@@ -86,108 +88,127 @@ func makeServiceConfig() config.Service {
 	}
-	if shouldFilterByNamespace(input.Istio.Enabled, input.Istio.Namespaces) {
-		processorName := getNamespaceFilterProcessorName(pipeline.Name, metric.InputSourceIstio)
-		cfg.Processors.NamespaceFilters[processorName] = makeFilterByNamespaceIstioInputConfig(pipeline.Spec.Input.Istio.Namespaces)
+	if isIstioInputEnabled(input) && shouldFilterByNamespace(input.Istio.Namespaces) {
+		processorID := makeNamespaceFilterID(pipeline.Name, metric.InputSourceIstio)
+		cfg.Processors.NamespaceFilters[processorID] = makeFilterByNamespaceIstioInputConfig(pipeline.Spec.Input.Istio.Namespaces)
 	}
-	if shouldFilterByNamespace(input.Otlp.Enabled, input.Otlp.Namespaces) {
-		processorName := getNamespaceFilterProcessorName(pipeline.Name, metric.InputSourceOtlp)
-		cfg.Processors.NamespaceFilters[processorName] = makeFilterByNamespaceOtlpInputConfig(pipeline.Spec.Input.Otlp.Namespaces)
+	if input.Otlp != nil && !input.Otlp.Disabled && shouldFilterByNamespace(input.Otlp.Namespaces) {
+		processorID := makeNamespaceFilterID(pipeline.Name, metric.InputSourceOtlp)
+		cfg.Processors.NamespaceFilters[processorID] = makeFilterByNamespaceOtlpInputConfig(pipeline.Spec.Input.Otlp.Namespaces)
 	}
+}
 
+func declareOTLPExporter(ctx context.Context, otlpExporterBuilder *otlpexporter.ConfigBuilder, pipeline *telemetryv1alpha1.MetricPipeline, cfg *Config, envVars otlpexporter.EnvVars) error {
 	otlpExporterConfig, otlpExporterEnvVars, err := otlpExporterBuilder.MakeConfig(ctx)
 	if err != nil {
 		return fmt.Errorf("failed to make otlp exporter config: %w", err)
 	}
 
 	maps.Copy(envVars, otlpExporterEnvVars)
 
-	otlpExporterID := otlpexporter.ExporterID(pipeline.Spec.Output.Otlp, pipeline.Name)
-	cfg.Exporters[otlpExporterID] = Exporter{OTLP: otlpExporterConfig}
-
-	pipelineID := fmt.Sprintf("metrics/%s", pipeline.Name)
-	cfg.Service.Pipelines[pipelineID] = makePipelineConfig(pipeline, otlpExporterID)
+	exporterID := otlpexporter.ExporterID(pipeline.Spec.Output.Otlp, pipeline.Name)
+	cfg.Exporters[exporterID] = Exporter{OTLP: otlpExporterConfig}
 
 	return nil
 }
 
-func makePipelineConfig(pipeline *telemetryv1alpha1.MetricPipeline, exporterIDs ...string) config.Pipeline {
-	sort.Strings(exporterIDs)
-
+func makeServicePipelineConfig(pipeline *telemetryv1alpha1.MetricPipeline) config.Pipeline {
 	processors := []string{"memory_limiter", "k8sattributes"}
 
 	input := pipeline.Spec.Input
-	if !*input.Runtime.Enabled {
+	if !isRuntimeInputEnabled(input) {
 		processors = append(processors, "filter/drop-if-input-source-runtime")
 	}
-	if !*input.Prometheus.Enabled {
+	if !isPrometheusInputEnabled(input) {
 		processors = append(processors, "filter/drop-if-input-source-prometheus")
 	}
-	if !*input.Istio.Enabled {
+	if !isIstioInputEnabled(input) {
 		processors = append(processors, "filter/drop-if-input-source-istio")
 	}
-	if !*input.Otlp.Enabled {
+	if input.Otlp != nil && input.Otlp.Disabled {
```

Comment:
can we also have a `isOtlpInputDisabled(input telemetryv1alpha1.MetricPipelineInput)` function?